### PR TITLE
fix: skip counting tabs without uri

### LINKED ISSUE #24: `tabsWithNumbers` doesn't work with search editors
Search editor has associated number but focusing command doesn't work

### PR DESCRIPTION
**Pull request by Ilanaya**
Closes #24

### CODE REVIEW

**Comment by zardoy** [comment]
Revert last commit please, it was working perfectly with > 20 schemes, supporting only `file` scheme breakes a lot of things (eg untitled or virtual files - this feature is supported on the web, <vscode.dev>)

**Comment by zardoy** [comment]
Also I just checked it, and it doesn't work properly, if search editor after first editor, focusing 2 doesn't focus 2. you forgot to refactor `focusTabFromLeft`

https://github.com/zardoy/vscode-experiments/issues/24#issuecomment-1242400677

> correct counting number: add in focusTabFromLeft and provideFileDecoration .filter before .findIndex, it should fix.

File: src/features/tabsWithNumbers.ts [hunk 1]
```diff
@@ -84,9 +85,9 @@ export default () => {
                     }
                 }
 
-                const tabIndex = (getExtensionSetting('showTabNumbers.reversedMode') ? [...recentFileStack].reverse() : recentFileStack).findIndex(
-                    elemUri => elemUri.toString() === uri.toString(),
-                )
+                const tabIndex = (getExtensionSetting('showTabNumbers.reversedMode') ? [...recentFileStack].reverse() : recentFileStack)
+                    .filter(({ scheme }) => scheme === 'file')
```
Comment (zardoy) [inline src/features/tabsWithNumbers.ts]: Why do we need filter `recentFileStack`?

1. I don't see a reason for this change
2. We can just not add uri to this array if it doesn't satisfy some requirement, which is also unclear

**Review by zardoy — CHANGES_REQUESTED** [review]

File: src/features/tabsWithNumbers.ts (hunk 1 above)
Comment (Ilanaya) [reply src/features/tabsWithNumbers.ts]: I agree filtering `recentFileStack` was bad idea, but the purpose of this change is removing file decoration from the `search-editor`, so I decided to skip search-editors from adding to `recentFIlestack` at all in https://github.com/zardoy/vscode-experiments/pull/26/commits/03601c726b0d2dc4795b9bcb76def69c7cb4e802.

**Comment by zardoy** [comment]
Looks good!

**Pull request merged**